optuna/samplers/_gp/sampler.py

```diff
@@ -289,7 +333,17 @@ def sample_independent(
         param_name: str,
         param_distribution: BaseDistribution,
     ) -> Any:
-        self._raise_error_if_multi_objective(study)
+        if len(study.directions) > 1:
+            warnings.warn(
+                f"Multi-objective optimization by {self.__class__.__name__} is an experimental "
+                "feature. The interface can change in the future."
+            )
```

Comment:
`GPSampler` itself is experimental. So, I don't think we need to add an optional experimental warning for the multi-objective optimization.